http/server.ts

```diff
@@ -232,7 +232,15 @@ async function readRequest(
     return [null, err];
   }
   [req.method, req.url, req.proto] = firstLine.split(" ", 3);
-  [req.headers, err] = await tp.readMIMEHeader();
+
+  // if there is an error in the request header
+  // we proceed the request but we force a 400
+  try {
+    [req.headers, err] = await tp.readMIMEHeader();
+  } catch (e) {
+    return [req, new Error("Unable to proceed request")];
```

Comment:
Wouldn't it be better to put the try...catch block around the `readRequest()` function call (so [here](https://github.com/denoland/deno_std/pull/419/files#diff-2e4c2adf52a4573569ea4cebb26b523eR266))?
I think there are more ways it can fail and they all require more or less the same treatment.

Reply:
i've added the `firstLine.split` in the try catch. the others statements are already pretty well handled.